server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -421,7 +408,7 @@ private UploadedMetadataResults writeMetadataInParallel(
                         blobStoreRepository.getCompressor(),
                         blobStoreRepository.getNamedXContentRegistry()
                     ),
-                    persistentSettingsBlobStore,
+                    remoteGlobalMetadataManager.getPersistentSettingsBlobStore(),
```

Comment:
why do you need to pass it at all if it is contained in the manager?

Reply:
If we don't pass the blobstore from here, we will need to add a if else again in `getAsyncMetadataWriteAction` to get which blobStore is needed for the remoteObject.